src/Dropdown.elm

```diff
@@ -328,14 +347,47 @@ closeOnlyIfNotMultiSelect config state =
     DropdownMsg subMsg ->
         let
             ( updated, cmd ) =
-                Dropdown.update dropdownConfig subMsg model.dropdownState model.items
+                Dropdown.update dropdownConfig subMsg model model.dropdownState model.items
         in
             ( { model | dropdownState = updated }, cmd )
 
 -}
-update : Config item msg -> Msg item -> State item -> List item -> ( State item, Cmd msg )
-update (Config config) msg (State state) data =
+update : Config item msg model -> Msg item -> model -> State -> List item -> ( State, Cmd msg )
+update (Config config) msg model (State state) data =
     let
+        selectedItems =
+            case config.selectionFromModel model of
+                SingleItem maybeItem ->
+                    Maybe.Extra.toList maybeItem
+
+                MultipleItems listItems ->
+                    listItems
+
+        modifySelectedItems selectedItem selectedItems_old =
+            case config.dropdownType of
+                MultiSelect ->
+                    -- if it was selected, we remove it from the list, otherwise include it
+                    if List.any ((==) selectedItem) selectedItems_old then
+                        List.filter ((/=) selectedItem) selectedItems_old
+
+                    else
+                        selectedItem :: selectedItems_old
+
+                _ ->
+                    [ selectedItem ]
+
+        updateSelectedItemsCommand selectedItems_new =
+            let
+                onSelectMsg =
+                    case config.onSelectMsg of
+                        OnSelectSingleItem itemToMsg ->
+                            itemToMsg <| List.head <| selectedItems_new
+
+                        OnSelectMultipleItems itemsToMsg ->
+                            itemsToMsg <| selectedItems_new
+            in
+            Task.succeed onSelectMsg |> Task.perform identity
```

Comment:
What do you think about extracting these 2 functions to one function like `selectedItemsCommand config selectedItems selectedItem`? this way we reduce the size of this function and make it a bit more meaningful. 
the function name it's just an example, if you find something better, feel free to rename it :D.

Also, we don't use much camel_case, so I would prefer to have all the functions/variables using pascal case.

Reply:
I personally find it clearer as two separate functions. But I've pulled them out of the let block into helper functions to reduce the size of update function, and added type annotations so hopefully they are more meaningful. Also removed the inconsistent _'s in variable names. https://github.com/PaackEng/elm-ui-dropdown/pull/24/commits/16c5f7449c2c72ba4e5c29222c6a87ed7f39786c